src/screens/SendSynthetic/SendSyntheticAsset.js

```diff
@@ -0,0 +1,239 @@
+import syntheticsService from 'services/synthetics';
+import { makePromiseCancelable } from 'utils/common';
+
+// constants
+import { SEND_TOKEN_CONTACTS } from 'constants/navigationConstants';
+import { PLR } from 'constants/assetsConstants';
+
+// selectors
+import { accountAssetsSelector } from 'selectors/assets';
+
+// models
+import type { Asset, Assets } from 'models/Asset';
+
+// configs
+import assetsConfig from 'configs/assetsConfig';
+import { availableStakeSelector } from 'selectors/paymentNetwork';
+
+type Props = {
+  accountAssets: Assets,
+  supportedAssets: Asset[],
+  initSyntheticsService: Function,
+  navigation: NavigationScreenProp<*>,
+  availableStake: number,
+};
+
+type State = {
+  loadingAssets: boolean,
+  availableAssets: [],
+};
+
+const InnerWrapper = styled(Wrapper)`
+  flex: 1;
+  margin-top: ${spacing.large}px;
+`;
+
+const ContentBackground = styled(Wrapper)`
+  flex: 1;
+  background-color: ${UIColors.defaultBackgroundColor};
+`;
+
+const genericToken = require('assets/images/tokens/genericToken.png');
+
+class SendSyntheticAsset extends React.Component<Props, State> {
+  state = {
+    loadingAssets: true,
+    availableAssets: [],
+  };
+  defaultSyntheticAsset: Asset;
+  syntheticServicePromise;
+
+  constructor(props: Props) {
+    super(props);
+    const { accountAssets, supportedAssets, availableStake } = props;
+    const assetsData = getAssetsAsList(accountAssets);
+    this.defaultSyntheticAsset = {
+      ...getAssetData(assetsData, supportedAssets, PLR),
+      balance: availableStake,
+      isSynthetic: true,
+    };
+  }
+
+  componentDidMount() {
+    this.props.initSyntheticsService();
+    this.getAvailableSyntheticsAssets();
+  }
+
+  componentWillUnmount() {
+    this.cancelSyntheticServiceRequest();
+  }
+
+  getAvailableSyntheticsAssets = () => {
+    const { supportedAssets, accountAssets } = this.props;
+    const assetsData = getAssetsAsList(accountAssets);
+    this.syntheticServicePromise = makePromiseCancelable(syntheticsService.getDataFromLiquidityPool());
+    this.syntheticServicePromise
+      .request()
+      .then((result) => {
+        const syntheticAssets = get(result, 'output.balanceResults', []);
+        const availableAssets = syntheticAssets.reduce((availableList, syntheticAsset) => {
+          const assetSymbol = get(syntheticAsset, 'token.symbol');
+          const assetBalance = Number(get(syntheticAsset, 'value', 0));
+          const assetData = getAssetData(assetsData, supportedAssets, assetSymbol);
+          if (!isEmpty(assetData) && assetBalance > 0) {
+            availableList.push({
+              ...assetData,
+              balance: assetBalance,
+              isSynthetic: true,
+            });
+          }
+          return availableList;
+        }, []);
+        this.setState({ availableAssets, loadingAssets: false });
+      })
+      .catch(({ isCanceled }) => {
+        if (!isCanceled) this.setState({ loadingAssets: false });
```

Comment:
What's the difference here if it was cancelled or just failed? Don't we want to do this regardless of if this was cancelled or not?

Reply:
The cancel part is because if it's triggered and we go back to previous screen and we don't want state to be updated in the component that is unmounted.

Reply:
Please mark as resolved if everything makes sense.